packages/devtools_app/lib/src/flutter/extent_delegate_list.dart

```diff
@@ -0,0 +1,459 @@
+import 'dart:math' as math;
+
+import 'package:collection/collection.dart' as collection;
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+/// Base class for delegate providing extent information for items in a list.
+abstract class ExtentDelegate {
```

Comment:
@kenzieschmoll I expect you will want to provide your own implementation of this class to leverage the fact that you know the absolute position of each element up front rather than really wanting to reason in terms of adding up sizes of each element. You might want to move some of the code in FixedExtentDelegate to a shared based class that handles binary searching for the matching row. The other difference is you probably ideally want to support gaps between rows. There isn't any reason this class couldn't support that although you will need to tweak the layout method as it currently assumes zero gaps so some of the logic is a bit odd if you have gaps.